Cython/Shadow.py

```diff
@@ -164,13 +164,13 @@ def cmod(a, b):
 
 def cast(t, *args, **kwargs):
     kwargs.pop('typecheck', None)
-    assert not kwargs
 
     if isinstance(t, typedef):
-        return t(*args)
+        assert not kwargs
+        return t(*args, **kwargs)
     elif isinstance(t, type):  # Doesn't work with old-style classes of Python 2.x
         if len(args) != 1 or not (args[0] is None or isinstance(args[0], t)):
-            return t(*args)
+            return t(*args, **kwargs)
 
     return args[0]
```

Comment:
I'm pretty sure the `assert not kwargs` was meant to make sure that no kwargs other than `typecheck` are used. Are there others now?

Reply:
I believe I changed this because if you look at the `cast` definition `**kwargs` are also forwarded. I can lie to the type checker though and type it as forwarding `**kwargs` ([because I cannot split them off](https://peps.python.org/pep-0612/#id1); see the `just_args` and `just_kwargs` examples) even though we error at runtime if any keyword arguments are passed?